azext_iot/tests/iothub/core/test_iothub_storage_int.py

```diff
@@ -321,49 +322,58 @@ def test_user_identity_storage(self):
+                self.cmd(
+                    'iot hub device-identity export -n {} --bcu "{}" --auth-type {} --identity {} --ik true'.format(
+                        self.entity_name, self.live_storage_uri, "identity", identity_id
+                    ),
+                    checks=[
+                        self.check("outputBlobContainerUri", self.live_storage_uri),
+                        self.check("failureReason", None),
+                        self.check("type", "export"),
+                        self.check("excludeKeysInExport", False),
+                        self.check("storageAuthenticationType", "identityBased"),
+                        self.exists("jobId"),
+                    ],
+                )
 
-        # give time to finish job
-        sleep(30)
+                # give time to finish job
+                sleep(30)
 
-        self.cmd(
-            'iot hub device-identity import -n {} --ibcu "{}" --obcu "{}" --auth-type {} --identity {}'.format(
-                self.entity_name, self.live_storage_uri, self.live_storage_uri, "identity", identity_id
-            ),
-            checks=[
-                self.check("outputBlobContainerUri", self.live_storage_uri),
-                self.check("inputBlobContainerUri", self.live_storage_uri),
-                self.check("failureReason", None),
-                self.check("type", "import"),
-                self.check("storageAuthenticationType", "identityBased"),
-                self.exists("jobId"),
-            ],
-        )
+                self.cmd(
+                    'iot hub device-identity import -n {} --ibcu "{}" --obcu "{}" --auth-type {} --identity {}'.format(
+                        self.entity_name, self.live_storage_uri, self.live_storage_uri, "identity", identity_id
+                    ),
+                    checks=[
+                        self.check("outputBlobContainerUri", self.live_storage_uri),
+                        self.check("inputBlobContainerUri", self.live_storage_uri),
+                        self.check("failureReason", None),
+                        self.check("type", "import"),
+                        self.check("storageAuthenticationType", "identityBased"),
+                        self.exists("jobId"),
+                    ],
+                )
 
-        self.cmd(
-            'iot hub device-identity export -n {} --bcu "{}" --auth-type {} --identity {}'.format(
-                self.entity_name, self.live_storage_uri, "identity", "fake_managed_identity"
-            ),
-            expect_failure=True
-        )
+                self.cmd(
+                    'iot hub device-identity export -n {} --bcu "{}" --auth-type {} --identity {}'.format(
+                        self.entity_name, self.live_storage_uri, "identity", "fake_managed_identity"
+                    ),
+                    expect_failure=True
+                )
+                user_identity_setup_completed = True
+            except Exception as x:
```

Comment:
Should we consider terminal exceptions vs retriable exceptions?

Reply:
I don't think it's needed for 2 reasons:

1. Most terminal exceptions would occur earlier in the test when the hub identity assignment happens (outside the try block)
![image](https://user-images.githubusercontent.com/51140335/132072597-38e1c0f6-1bf9-4aaf-8fbb-1e6317a5ffb0.png)

2. We are only attempting 2 extra times, which will not waste too much time in case there are any other terminal exceptions occur